auth.c

```diff
@@ -955,8 +955,18 @@ subprocess(const char *tag, struct passwd *pw, const char *command,
 		if (posix_spawn_file_actions_init(&actions) != 0 ||
 			posix_spawn_file_actions_adddup2(&actions, p[1], STDOUT_FILENO) != 0)
 			fatal("posix_spawn initialization failed");
-		else if (__posix_spawn_asuser((pid_t*)&pid, av[0], &actions, NULL, av, NULL, pw->pw_name) != 0)
-			fatal("posix_spawn: %s", strerror(errno));
+		else {
+			if (strcmp(pw->pw_name, "system") == 0 && am_system()) {
```

Comment:
@bkatyl - This mayn't succeed on a windows machine with non-english culture. It's better to get the SID and compare the SID with WinLocalSystemSid. 
Please refer to the logic [here](https://github.com/PowerShell/openssh-portable/blob/1a1a2ac5f1720f791416f3f16e15b369a5f05fe9/contrib/win32/win32compat/w32-sshfileperm.c#L122)

Reply:
Thanks for the feedback and pointer to the logic, it was very helpful. 

I added a commit that includes the changes to SID comparison. Since these are windows only calls, I moved the evaluation into the Windows ifdef. The needed imports are also guarded by a Windows ifdef.